api/src/main/java/jakarta/enterprise/inject/spi/BeanManagerLite.java

```diff
@@ -0,0 +1,246 @@
+     * @throws AmbiguousResolutionException if the ambiguous dependency resolution rules fail
+     * @throws IllegalStateException        if called during application initialization, before the {@link AfterBeanDiscovery}
+     *                                      event is fired.
+     */
+    <X> Bean<? extends X> resolve(Set<Bean<? extends X>> beans);
+
+    /**
+     * Return an ordered set of {@linkplain ObserverMethod observer methods} for an event.
+     * <p>
+     * Note that when called during invocation of an {@link AfterBeanDiscovery} event observer,
+     * this method will only return observers discovered by the container before the {@link AfterBeanDiscovery} event is fired.
+     *
+     * @param <T>        the type of the event
+     * @param event      the event object
+     * @param qualifiers the event qualifiers
+     * @return the resulting set of {@linkplain ObserverMethod observer methods}
+     * @throws IllegalArgumentException if the runtime type of the event object contains a type variable
+     * @throws IllegalArgumentException if two instances of the same non repeating qualifier type are given
+     * @throws IllegalArgumentException if an instance of an annotation that is not a qualifier type is given
+     * @throws IllegalStateException    if called during application initialization, before the {@link AfterBeanDiscovery}
+     *                                  event is fired.
+     */
+    <T> Set<ObserverMethod<? super T>> resolveObserverMethods(T event, Annotation... qualifiers);
+
+    /**
+     * Return an ordered list of enabled {@linkplain Interceptor interceptors} for a set of interceptor bindings and a type of
+     * interception and which are enabled in the module or library containing the class into which the <code>BeanManager</code> was
+     * injected or the Java EE component from whose JNDI environment namespace the <code>BeanManager</code> was obtained.
+     * <p>
+     * Note that when called during invocation of an {@link AfterBeanDiscovery} event observer,
+     * this method will only return interceptors discovered by the container before the {@link AfterBeanDiscovery} event is
+     * fired.
+     *
+     * @param type                the type of the interception
+     * @param interceptorBindings the interceptor bindings
+     * @return the resulting set of {@linkplain Interceptor interceptors}
+     * @throws IllegalArgumentException if no interceptor binding type is given
+     * @throws IllegalArgumentException if two instances of the same interceptor binding type are given
+     * @throws IllegalArgumentException if an instance of an annotation that is not an interceptor binding type is given
+     * @throws IllegalStateException    if called during application initialization, before the {@link AfterBeanDiscovery}
+     *                                  event is fired.
+     */
+    List<Interceptor<?>> resolveInterceptors(InterceptionType type, Annotation... interceptorBindings);
```

Comment:
I was also wondering -- do we actually want these 2 methods (`resolveObserverMethods` and `resolveInterceptors`)? I know we implement them in Quarkus, but they sound a bit too runtime-oriented.